tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/json/FirebaseSubscriptionSerializer.scala

```diff
@@ -35,15 +39,44 @@ object FirebaseSubscriptionSerializer {
         "expires" -> response.expires,
         "types" -> response.types)
 
+  private implicit val mapCreationRequestBySubscriptionCreationId: Reads[Map[FirebaseSubscriptionCreationId, JsObject]] =
+    Reads.mapReads[FirebaseSubscriptionCreationId, JsObject] { string =>
+      refineV[IdConstraint](string)
+        .fold(e => JsError(s"firebase subscription id needs to match id constraints: $e"),
+          id => JsSuccess(FirebaseSubscriptionCreationId(id)))
+    }
+
   private implicit val idFormat: Format[UnparsedFirebaseSubscriptionId] = Json.valueFormat[UnparsedFirebaseSubscriptionId]
   private implicit val firebaseSubscriptionIdsReads: Reads[FirebaseSubscriptionIds] = Json.valueReads[FirebaseSubscriptionIds]
   private implicit val firebaseSubscriptionGetResponseWrites: Writes[FirebaseSubscriptionGetResponse] = Json.writes[FirebaseSubscriptionGetResponse]
   private implicit val firebaseSubscriptionIdsWrites: Writes[FirebaseSubscriptionIds] = Json.writes[FirebaseSubscriptionIds]
 
   private implicit val firebaseSubscriptionGetRequestReads: Reads[FirebaseSubscriptionGetRequest] = Json.reads[FirebaseSubscriptionGetRequest]
 
+  private implicit val firebaseSubscriptionSetRequestReads: Reads[FirebaseSubscriptionSetRequest] = Json.reads[FirebaseSubscriptionSetRequest]
+  private implicit val firebaseSubscriptionCreationResponseWrites: Writes[FirebaseSubscriptionCreationResponse] = Json.writes[FirebaseSubscriptionCreationResponse]
+  private implicit val subscriptionMapSetErrorForCreationWrites: Writes[Map[FirebaseSubscriptionCreationId, SetError]] =
+    mapWrites[FirebaseSubscriptionCreationId, SetError](_.serialise, setErrorWrites)
+
+  private implicit val subscriptionMapCreationResponseWrites: Writes[Map[FirebaseSubscriptionCreationId, FirebaseSubscriptionCreationResponse]] =
+    mapWrites[FirebaseSubscriptionCreationId, FirebaseSubscriptionCreationResponse](_.serialise, firebaseSubscriptionCreationResponseWrites)
+
+  private implicit val firebaseSubscriptionSetResponseWrites: OWrites[FirebaseSubscriptionSetResponse] = Json.writes[FirebaseSubscriptionSetResponse]
+  private implicit val subscriptionExpiredTimeReads: Reads[FirebaseSubscriptionExpiredTime] = Json.valueReads[FirebaseSubscriptionExpiredTime]
```

Comment:
Use `valueFormat instead, to combine valueReads and valueWrites in one line.

Please apply this comment where relevant.